cfpb/sbl-frontend
# [feat] 404 - Not Found

# WIP - NOT READY FOR REVIEW

Closes #129 

## Pending questions
- Do we want all the same/similar links that we see on the [cfgov 404 page](https://www.consumerfinance.gov/1/).  As currently implemented, the links/buttons in the screenshot below would direct users to:
  - `Platform homepage` -> Unauthenticated landing page
  - `Contact us` -> Currently links to an unimplemented `/about-us/contact-us/` page.  Maybe this should link to SBL Help?
  - `Let us know` -> Opens an email addressed to `CFPB_website@consumerfinance.gov` with info about the current, unrecognized URL, matching how [cfgov's 404 page](https://www.consumerfinance.gov/1/) handles it. 

## Changes
- Adds a `404 - Not Found` page that is displayed for any unrecognized application routes
- Addresses some `no magic number` eslint errors that were blocking code commit
- Mobile styles are included in the content that was copied from CF.gov 🎉 

## How to test this PR

1. Visit http://localhost:8899/not-found
2. Observe the 404 page
3. http://localhost:8899/institution/TESTBANK123
4. Observe app content

## Screenshots

### Desktop
![Image](https://github.com/cfpb/sbl-frontend/assets/2592907/233f8eb3-eb2e-432e-a350-685a22ce0240)

### Mobile
![screencapture-localhost-8899-about-us-contact-us-2024-01-08-16_24_42](https://github.com/cfpb/sbl-frontend/assets/2592907/5373518f-49cb-40d0-862b-3d2719132df4)

## Notes

Since the CF.gov file included styles for a `500 - Server error page` in the same style file, I also copied those required image assets, though I don't believe our react app would be able to render a `500` page - that would probably need to be from NGINX or some other layer of our tech stack.  

...And after pondering the above, I've removed the `500 - Server error page` related assets/styles.
